Create Meeting Notes 50 (#312)
cardano-foundation/CIPs: BiweeklyMeetings/2022-08-02.md

**Last Check**: Review of the PRIOR meetings Decisions  - if no objection, apply change (effectively a two week lag from decision to action, as a grace period)  

**New CIPs Review**: CIPs up for review should be looked over collectively, with discussion where needed. (on top of the asynchronous reviews)  

PR -> ‘Draft’: Needs format + approval.  
‘Draft’ -> ‘Proposed’: Needs a PLAN towards Active + implementation.  
‘Proposed’ -> ‘Active’:  Objective criteria as laid out observed, and consensus agreeing.   
**Current Discussions**: What the current CIPs discussions are on social media / forums / Discord.  

**Issues**: Review of the open Issues on the CIP GitHub 

**Close**: Recap of actions taken and decisions. List the CIPs that are due for review.  

## August 02 2022 Transcript
**Attending Editors**: Matthias Benkort, Sebastien Guillemot, ~Frederic Johnson~, ~Robert Phair~. 
**Guests**: ~Morgan Thomas~, Michael Liesenfelt

### Triage

#### PR214
[PR214 CIP-0027? | Fix status header](https://github.com/cardano-foundation/CIPs/pull/214)

> **Matthias** -: Yes. So we're looking into that one. It's a pretty old one, I think. Yeah, ninth, February. It was just changing the status of a CIP number 27, which I think is... And debated here, not the CIP, but the fact that this is an active CIP nowadays, there have been people disagreeing with that standard and coming up with new standards, which is also fine. But as far as we are concerned, it's indeed active. So there is no problem with the change for me. Robert was approved. The only thing is we will also need to update the [inaudible 00:07:23], with that to reflect it. So we can do this as part of a recent PI updated to update the rhythm and yeah, just get to move on with this particular one, unless you, I don't know, have something to say Sebastien. But-

Reviewer: ```suggestion
**Matthias** -: Yes. So we're looking into that one. It's a pretty old one, I think. Yeah, ninth, February. It was just changing the status of a CIP number 27, which I think is... And debated here, not the CIP, but the fact that this is an active CIP nowadays, there have been people disagreeing with that standard and coming up with new standards, which is also fine. But as far as we are concerned, it's indeed active. So there is no problem with the change for me. Robert has approved. The only thing is we will also need to update the [inaudible 00:07:23], with that to reflect it. So we can do this as part of a recent PI updated to update the rhythm and yeah, just get to move on with this particular one, unless you, I don't know, have something to say Sebastien. But-
```

**Matthias** -: Let me also watch the stream. Yeah. Well I guess that's one of [inaudible 00:08:38] we have no news from the author, probably came from a discussions. They made a change to the CIP 13, I think Sebastian response is...

**Sebastien** -: Yeah. Because the problem with this one is that the way they're doing it is different from the way eternal did this.

**Matthias** -: Yeah.

**Sebastien** -: So somebody needs to decide how they want to resolve this mismatch.

**Matthias** -: So do you think we should pull not selling the discussions or-

**Sebastien** -: Yeah. Trying paying, we can try and paint them or sell it and see if they want to resolve this or if they want to drop the CIP. They need to figure this out.

**Matthias** -: Yeah. I mean, since there has been no activity for four months on that one, I would assume it's probably being dropped. So I would just ping, Oslin and maybe Marcel to see if they have anything to comment.

**Matthias** -: Okay. And I will just mark it as waiting for author.

**Anthony**-: Amazing. Excellent. So is everybody happy to move on to CIP 45?

> **Matthias** -: Yes. Yes. Now just make sure we keep CIP 30 for let's check, maybe next meeting. And if no activity has happened, then we'll just I think close it.

Reviewer: ```suggestion
**Matthias** -: Yes. Yes. Now just make sure we keep CIP 30 for last check, maybe next meeting. And if no activity has happened, then we'll just I think close it.
```

**Anthony**-:Okay. Excellent. I'll do so. Great. Thank you, Matthias.

#### PR229
[PR229 CIP-0045? | Decentralization: Using Pledge as a Bidding Param](https://github.com/cardano-foundation/CIPs/pull/229)

> **Matthias** -:So CIP 45. Yeah. So I had a quick look at this one this morning and also at the forums messages. So this is yet another CIP to address incentives, rewards, and things that are in that realm, specifically focused on the pledge, Stake Pool pledge that one. So it sounded a lot similar to a CIP that was discussed in the past, which was the dynamic saturation based on pledge. So the idea of CIP 45 is to make the pledge for pros also dynamic in a way, and depends on the actual number of ADA being stable, the circulating supply, and that way make sure that both the pledge and the saturation levels kind of reflects the actual state of the network. So I haven't been through all the equations just yet, only been through the comments on the forum.

Reviewer: pledge for ~pros~pools also dynamic ???

**Matthias** -: So yeah, I think it's a good one to put under review as well. And I would be interested to see a discussion comparing that with CIP 37, and how did you know defers on that. I'm pretty sure both authors know or that the two [inaudible 00:12:30] exist, if they don't and it's a good thing we put them together maybe for a chat as well.

**Anthony**-:Okay, excellent.So... Apologies. So from my perspective, this is something that's going to be placed under review.

**Matthias** -: Yeah. I've been also quite late to address that one in a way, because I see it's been published on the 1st of March, so it's been sitting there for quite a bit, might be a bit of effort to resurrect the conversation, but I would definitely yes, love to spend some more time to go through this one point by point, maybe have a kind of comparisons of that with CIP 37. I'm not sure if you had time to go through it, Sebastien, if you have any remarks as well on this one.

**Sebastien** -: No, I don't actually. And do you know why this one was missed for so long? Did we talk about this? And then there was a reason why we lost over it.

> **Matthias** -: And I think it just got lost in translation, asleep for the cracks. So I don't know. That's kind of CIP also wants to get IOG attentions about, and we know it's been a difficult task. We could probably also pass it through Reza at the CF, is also now getting to more and more into looking into these things, just at least to get some first also research feedback on this. And I will also ping Kevin HAmond about that, who is now our point of contact with IOG and our best chance to get something through IOG research.

Reviewer: ```suggestion
**Matthias** -: And I think it just got lost in translation, asleep for the cracks. So I don't know. That's kind of CIP also wants to get IOG attentions about, and we know it's been a difficult task. We could probably also pass it through Reza at the CF, is also now getting to more and more into looking into these things, just at least to get some first also research feedback on this. And I will also ping Kevin Hammond about that, who is now our point of contact with IOG and our best chance to get something through IOG research.
```

**Michael**-: To become the best version of itself.

**Matthias** -: Yes. Let's maybe quote that under the CIP title.

**Matthias** -: Okay. So let's keep the discussions going also and have CIP 50 back for review next week. We haven't got any updates from IOG, think they have been busy with Vasil with some of the recent problem and [inaudible 00:35:06] found. So I haven't heard of Kevin on that. I think Christoff mentioned that he... I mean Kevin said something recently online on the 360 or something like that, but it was nothing really substantial, just that they were looking at it. They were aware that it existed, wanted to well provide also [inaudible 00:35:30]. We'll still a bit waiting on that sentence. So let's see next meeting.

**Anthony**-: Amazing. Great.

#### PR296
[PR296 CIP-0062? | Governance API for dApp Connectors (dApps<->Wallet Web-bridge](https://github.com/cardano-foundation/CIPs/pull/296)

**Matthias** -: CIP 62 is next in line, right?

**Anthony**-: That's correct.

> **Matthias** -: I've been trying to click on the link for 15 seconds and I just realized it was a stream. So can click on the video, obviously. Governance API for DApp, which was proposed by [inaudible 00:36:17] Martins, who is now the tech lead regarding Wallets at IOG correctly. And this is a proposal to extend CIP 30 with capabilities that would make that voting process a bit more streamlined. Sorry, I did a first run of [inaudible 00:36:42] on that one this morning and mainly suggested to simplify a bit some of the methods in particular to make things a bit more [inaudible 00:36:54], meaning not to have the API provide five different ways of doing the same things or in that case two different ways.

Reviewer 2: ```suggestion
**Matthias** -: I've been trying to click on the link for 15 seconds and I just realized it was a stream. One can't click on a video, obviously. Anyway, governance API for DApp, which was proposed by Bruno Martins, who is now the tech lead regarding Wallets at IOG if I recall correctly. And this is a proposal to extend CIP 30 with capabilities that would make that voting process a bit more streamlined. Sorry, I did a first run of review on that one this morning and mainly suggested to simplify a bit some of the methods in particular to make things a bit more orthogonal, meaning not to have the API provide five different ways of doing the same things or in that case two different ways.
```

> **Matthias** -: And I'm also asking for a rational section mainly because we need a rational section for any CIPs in general, because that what exists the discussion for all the bases for discussion and that explains, at a later time, while the decisions were made, the waiting were made. And for that one in particular, because there are a few suggested API or endpoints, you could say, or interfaces, that can be done currently through existing libraries and I don't see why adding them to the bridge, the wallet interface, so to speak makes sense. Or at least there is no justification for it. I can get that it's maybe a bit more simple and that removes some of the burden from the apps implementing that, but those apps are mainly Wallets anyway. So that was a bit some form of the gap in the justification here that I'm liking. I don't know, Sebastien, if you had a look at that yet.

Reviewer: ```suggestion
**Matthias** -: And I'm also asking for a rationale section mainly because we need a rationale section for any CIPs in general, because that what exists the discussion for all the bases for discussion and that explains, at a later time, while the decisions were made, the waiting were made. And for that one in particular, because there are a few suggested API or endpoints, you could say, or interfaces, that can be done currently through existing libraries and I don't see why adding them to the bridge, the wallet interface, so to speak makes sense. Or at least there is no justification for it. I can get that it's maybe a bit more simple and that removes some of the burden from the apps implementing that, but those apps are mainly Wallets anyway. So that was a bit some form of the gap in the justification here that I'm liking. I don't know, Sebastien, if you had a look at that yet.
```
"the waiting were made" = ???

> **Matthias** -: And I'm also asking for a rationale section mainly because we need a rationale section for any CIPs in general, because that what exists the discussion for all the bases for discussion and that explains, at a later time, while the decisions were made, the waiting were made. And for that one in particular, because there are a few suggested API or endpoints, you could say, or interfaces, that can be done currently through existing libraries and I don't see why adding them to the bridge, the wallet interface, so to speak makes sense. Or at least there is no justification for it. I can get that it's maybe a bit more simple and that removes some of the burden from the apps implementing that, but those apps are mainly Wallets anyway. So that was a bit some form of the gap in the justification here that I'm liking. I don't know, Sebastien, if you had a look at that yet.

Reviewer 2: ```suggestion
**Matthias** -: And I'm also asking for a rationale section mainly because, we need a rationale section for any CIPs in general, and because that gives the bases for discussions and that explains, at a later time, why the decisions were made, the context within which they were made. And for that one in particular, because there are a few suggested API or endpoints, you could say, or interfaces, that can be done currently through existing libraries and I don't see why adding them to the bridge, the wallet interface so to speak, makes sense. Or at least there is no justification for it. I can get that it's maybe a bit more simple and that removes some of the burden from the apps implementing that, but those apps are mainly Wallets anyway. So that was a bit some form of the gap in the justification here that I'm liking. I don't know, Sebastien, if you had a look at that yet.
```

**Sebastien** -: No, unfortunately I was hoping to look into it today, but as you may have heard, there's other stuff that came up. There's still some discussion going on internally related to the CIP, so stuff may still change, but definitely it's good to over it. I agree that it's probably over complicated the way it's testified now and I'll try and find some time to give my thoughts on this.

> **Matthias** -: Okay. Because my main understanding on this, or I mean, experience that's... I mean the catalyst voting in general, it's always been done by [inaudible 00:39:01] themselves. So as part of the Wallet interface, usually do things a bit differently that usually Turmo for that in the past, it's pretty nice just acreage transaction registration as part of the Wallet itself. So I don't quite see the need for an external API for that to let any depth to the registration themselves.

Reviewer 2: ```suggestion
**Matthias** -: Okay. Because my main understanding on this, or I mean, experience that's... I mean the catalyst voting in general, it's always been done by wallet implementors themselves; [Inaudible] So I don't quite see the need for an external API for that to let any DApp do the registration themselves.
```

**Matthias** -: Okay. So, then I don't get the API cause a lot of the API are-

**Sebastien** -: Yeah.

**Matthias** -: It's leveraging key derivation path, which suggests that the key belongs to the Wallet and not to some delegated-

**Sebastien** -: Yeah. I think this is a mistake. It should be like other standards where you start with zero and then if zero is huge to get one and the Wallet handles this, I think. Having the debt analysis is going to be a huge disaster.

**Matthias** -: So okay. Definitely a need for some of the motivation and rational. So I've already flagged that on the PR waiting for some comments now on this. Yeah. So I guess we can just leave it there for now. If we have any of update from the author before the next meeting, we can discuss that again in the next meeting otherwise, I just wait.

#### PR299
[PR299 CIP-0068? | Datum Metadata Standard](https://github.com/cardano-foundation/CIPs/pull/299)

> **Matthias** -: To do. Next one on the agenda, CIP 68. Day two metadata standard, which comes as a replacement or an iteration on the existing metadata standards. But this one aims to be a bit general, although the specification is also focused on entities at this stage because that's the main use case driving that proposal.

Reviewer: ```suggestion
**Matthias** -: To do. Next one on the agenda, CIP 68. Datum metadata standard, which comes as a replacement or an iteration on the existing metadata standards. But this one aims to be a bit general, although the specification is also focused on entities at this stage because that's the main use case driving that proposal.
```
"Day two" metadata... sounds unappetising 🤣

> **Matthias** -: To do. Next one on the agenda, CIP 68. Datum metadata standard, which comes as a replacement or an iteration on the existing metadata standards. But this one aims to be a bit general, although the specification is also focused on entities at this stage because that's the main use case driving that proposal.

Reviewer 2: ```suggestion
**Matthias** -: To do. Next one on the agenda, CIP 68. Datum metadata standard, which comes as a replacement or an iteration on the existing metadata standards. But this one aims to be a bit general, although the specification is also focused on NFTs at this stage because that's the main use case driving that proposal.
```

> **Matthias** -: The main spirit as for... I think the main rationale as for why put a metadata in the datum instead of in the transaction is because you cannot access metadata of the transaction as part of your produce script, which is a little odd. And it truly sounds like at a political level, at least the transaction metadata and the data room have become kind of redundant with one another. So having access to the metadata inside the script allows for some extra validations to be done in particular when moving NFTs around.

Reviewer 2: ```suggestion
**Matthias** -: The main spirit as for... I think the main rationale as for why put a metadata in the datum instead of in the transaction is because you cannot access metadata of the transaction as part of your Plutus script, which is a little odd. And it truly sounds like at a political issue, at least the transaction metadata and the datums have become kind of redundant with one another. So having access to the metadata inside the script allows for some extra validations to be done in particular when moving NFTs around.
```

> **Matthias** -: The main spirit as for... I think the main rational as for why put a metadata in the data room instead of in the transaction is because you cannot access metadata of the transaction as part of your produce script, which is a little odd. And it truly sounds like at a political level, at least the transaction metadata and the data room have become kind of redundant with one another. So having access to the metadata inside the script allows for some extra validations to be done in particular when moving NFTs around.

Reviewer: ```suggestion
**Matthias** -: The main spirit as for... I think the main rationale as for why put a metadata in the datum instead of in the transaction is because you cannot access metadata of the transaction as part of your produce script, which is a little odd. And it truly sounds like at a political level, at least the transaction metadata and the data room have become kind of redundant with one another. So having access to the metadata inside the script allows for some extra validations to be done in particular when moving NFTs around.
```
Strangely "put a metadata in the data room" actually makes sense here 🤣

> **Matthias** -: Without access to that, your scripts are basically blind to what happened, which makes it very hard to just update metadata, for instance, if that's ever needed. So I think that's the core motivations behind that CIP. Beyond that the structure is kind of similar to what was already proposed for the NFT standards, except that it's fully [inaudible 00:42:52] based now and something that is meant to be provided as [inaudible 00:42:58] as produce built in data and possibly interpreted as Jason once again. And well, that's about what I have for now on that. There have been a lot of discussions on it. I admittedly, didn't go through all the chats yet. Sebastien, I see you've been a bit more active than me on this one. So maybe you have some inputs as well.

Reviewer 2: ```suggestion
**Matthias** -: Without access to that, your scripts are basically blind to what happened, which makes it very hard to just update metadata, for instance, if that's ever needed. So I think that's the core motivations behind that CIP. Beyond that the structure is kind of similar to what was already proposed for the NFT standards, except that it's fully accessible from the script's context and something that is meant to be provided as datums, so Plutus built-in data, and possibly interpreted as JSON once again. And well, that's about what I have for now on that. There have been a lot of discussions on it. I admittedly, didn't go through all the comments yet. Sebastien, I see you've been a bit more active than me on this one. So maybe you have some inputs as well.
```

> **Matthias** -: Okay. We haven't really looked at CIP 66. So this did stand up proposed by the team, which might be good then to bring on the table, at least in the next meeting or next to the new metadata discussion, especially if they have some conflicting opinions on these.

Reviewer 2: ```suggestion
**Matthias** -: Okay. We haven't really looked at CIP 66. So this DID standard proposed by the team, which might be good then to bring on the table, at least in the next meeting or next to the new metadata discussion, especially if they have some conflicting opinions on these.
```

**Sebastien** -: Yeah, I think there were three PRs with conflicting implementations of this many data standard. So there's the metadata standard for us as the end maker one. Then there's a third one that also had something similar. I don't remember.

**Matthias** -: Well, there was on chain metadata proposed by this web team a while back, but I think this one got...

**Sebastien** -: Cardano Smart NFTs.

**Matthias** -: Oh no, that was yet another one, the smart NFTs. The smart NFT were about embedding actual codes on the blockchains that makes those NFT... I mean, that's an API for those NFTs to be embedded.
Speaker 4:
Right. And they want the way to upgrade these...

> **Matthias** -: Yeah. To make them [inaudible 00:46:06] and interactive in a way. But there is some overlap maybe less than with CIP six. Okay. So that would be important to mention and to see if we can get maybe both offers to discuss where there is no problem in having two different proposals. That's always a pain down the line. So platform that need to implement both of them as that both gets really adopted. So we might as well want to have only one. So I'll mention that on the request.

Reviewer 2: ```suggestion
**Matthias** -: Yeah. To make them programmable and interactive in a way. But there is some overlap maybe less than with CIP 66. Okay. So that would be important to mention and to see if we can get maybe both offers to discuss where there is no problem in having two different proposals. That's always a pain down the line. So platform that need to implement both of them as that both gets really adopted. So we might as well want to have only one. So I'll mention that on the request.
```

**Matthias** -: Yeah, I'm just finishing up a comment on CIP 68. I can do a small summary of the session after that. No problem. Just trying to get the handle for the author.

### Discussions
N/A

### Issues
N/A

### Close 

**Matthias** -: Okay. So back to the meeting and the agenda. So we've been through, well CIP 27 of the status of the exchange with move it to active.

**Matthias** -: Then CIP 30, there was the discussion of adding a Wallet experimental API version to it, which happened while back has been idle since. There was a suggestion from Sebastien a while back also to use the Wallet name, to address what that particular suggestion was about. So waiting for the author and we'll close it in the next biweekly if there is no update on that one.

**Matthias** -: CIP 45, which is another proposal that tries to address the pledge/incentives issues on Cardano. We somehow completely missed that one in the past because it's been there since the 1st of March. So we're trying to get back to the author now and see if we can spark back the discussions on this. We reflect this to you as well, to see if we can get that into their feedback pipeline.

> **Matthias** -: We'll discuss also updates on CIP 50, mainly related to the recent blog posts that were shared by IOG and the research team. A few action items on that being, make a response to those blog posts and as well, update some of the graphs to take a slightly different perspective. Then what we currently have. And there was also these discussions of maybe trying to make some of the data available to wallets or applications, which also happened on this call as well. We discussed briefly CIPP 62, which is an extension of CIP 30 to add methods regarding the catalyst voting registration. It wasn't clear as for why one would want that as part of the CIP 30, that connector. And so if like that, asking for a rational or some justifications from the original author.

Reviewer: ```suggestion
**Matthias** -: We'll discuss also updates on CIP 50, mainly related to the recent blog posts that were shared by IOG and the research team. A few action items on that being, make a response to those blog posts and as well, update some of the graphs to take a slightly different perspective. Then what we currently have. And there was also these discussions of maybe trying to make some of the data available to wallets or applications, which also happened on this call as well. We discussed briefly CIP 62, which is an extension of CIP 30 to add methods regarding the catalyst voting registration. It wasn't clear as for why one would want that as part of the CIP 30, that connector. And so if like that, asking for a rationale or some justifications from the original author.
```

> **Matthias** -: And finally CIP 68 regarding the data metadata standard, which comes as a replacement of CIP 25, there seems to be some overlap with CIP 66. They have both different, slightly different trade offs. So we are suggesting to have a discussion between the authors of the two CIPs and so have that your discussion next week. And that's about it.

Reviewer: ```suggestion
**Matthias** -: And finally CIP 68 regarding the datum metadata standard, which comes as a replacement of CIP 25, there seems to be some overlap with CIP 66. They have both different, slightly different trade offs. So we are suggesting to have a discussion between the authors of the two CIPs and so have that your discussion next week. And that's about it.
```
